### PR TITLE
IIA-21-Terms and conds updated

### PR DESCRIPTION
**Pull request by SaniutaRobert**
Simulator Screen Shot - iPhone 11 - 2021-11-05 at 13 34 40

### CODE REVIEW

**Review by GeorgeCuciureanu — APPROVED** [review]

**Comment by SaniutaRobert** [comment]
Va fii integrat cu Logic Star Implemented

File: EnMovie/Entity/Favorites/FavoriteMovies.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+struct FavoriteMovies: Codable, Equatable {
+    var id : Int
+    var title : String
+    var genre_ids : [Int]
+    var release_date : String
+    var poster_path : String
+    var vote_average : Double
+}
+
+struct CategoriesResults: Decodable{
```
Comment (IgnatCamelia) [inline EnMovie/Entity/Favorites/FavoriteMovies.swift:25]: Space before {

File: EnMovie/Entity/Favorites/FavoriteMovies.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    var genres : [Categories]
+}
+
+struct Categories : Codable {
+    var id: Int
+    var name: String
+}
+
+struct FinalMovieStruct{
```
Comment (IgnatCamelia) [inline EnMovie/Entity/Favorites/FavoriteMovies.swift:34]: Space before {

File: EnMovie/Entity/Favorites/FavoritesDataProvider.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import Foundation
+import UIKit
+
+protocol FavoritesDataProviderProtocol {
+    func getFavMoviesData (completion: @escaping (Result<[FavoriteMovies], Error>) -> Void)
+    func getCategories(completion : @escaping ([Categories]) -> ())
+}
+
+struct Poster : Decodable{
```
Comment (IgnatCamelia) [inline EnMovie/Entity/Favorites/FavoritesDataProvider.swift:16]: Space before {

File: EnMovie/Entity/Favorites/FavoritesDataProvider.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    var image : String
+}
+
+class FavoritesDataProvider : FavoritesDataProviderProtocol{
```
Comment (IgnatCamelia) [inline EnMovie/Entity/Favorites/FavoritesDataProvider.swift:20]: Space before {

File: EnMovie/Entity/Stars/StarLogic.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            if residue != 0.0 {
+                let imageView = UIImageView()
+                imageView.image = UIImage(systemName: "star.fill.left")
+                imageView.tintColor = UIColor.yellow
+                stackView.addArrangedSubview(imageView)
+            }
+            let label = UILabel()
+            label.text = " \(numberOfStars)/5"
+            stackView.addArrangedSubview(label)
+        }else{
```
Comment (IgnatCamelia) [inline EnMovie/Entity/Stars/StarLogic.swift:36]: Space after } and before {

File: EnMovie/Modules/Favorites/View/FavoritesViewController.swift [hunk 6]
```diff
@@ -17,13 +21,73 @@ protocol FavoritesViewOutput {
...
+            self.favoritesTableView.reloadData()
+        }
+    }
 }
+
+extension FavoritesViewController : UITableViewDelegate,UITableViewDataSource{
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if moviesData.isEmpty {
+            return 1
+        }else{
```
Comment (IgnatCamelia) [inline EnMovie/Modules/Favorites/View/FavoritesViewController.swift:61]: Space after } and before {

File: EnMovie/Modules/Favorites/View/FavoritesViewController.swift [hunk 7]
```diff
@@ -17,13 +21,73 @@ protocol FavoritesViewOutput {
...
+                tableView.isHidden = true
+                dataAndMessageLabel.text = "You have no movies added"
+                spinner.stopAnimating()
+                spinner.isHidden = true
+            }else{
+                tableView.isHidden = true
+                dataAndMessageLabel.text = "Data is loading"
+                spinner.startAnimating()
+            }
+        }else{
```
Comment (IgnatCamelia) [inline EnMovie/Modules/Favorites/View/FavoritesViewController.swift:80]: Same

File: EnMovie/Modules/Favorites/View/FavoriteCell.swift [hunk 8]
```diff
@@ -0,0 +1,53 @@
+//
+//  FavoriteCell.swift
+//  EnMovie
+//
+//  Created by Robert Saniuta on 28.10.2021.
+//
+
+import UIKit
+
+
```
Comment (IgnatCamelia) [inline EnMovie/Modules/Favorites/View/FavoriteCell.swift:10]: Remove unnecessary spaces

**Pull request closed without merging**